src/Microsoft.Diagnostics.EventFlow.Inputs.NLog/NLogInput.cs

```diff
@@ -0,0 +1,265 @@
+            // Prefer the built-in `Message` and `Exception` properties by adding them to the payload
+            // first. If other attached data items have conflicting names, they will be added as
+            // `Message_1` and so-on.
+            if (logEvent.Exception != null)
+            {
+                if (logEvent.Level.Ordinal >= NLog.LogLevel.Error.Ordinal)
+                {
+                    EventMetadata eventMetadata = new EventMetadata(ExceptionData.ExceptionMetadataKind);
+                    eventMetadata.Properties.Add(ExceptionData.ExceptionPropertyMoniker, "Exception");
+                    eventData.SetMetadata(eventMetadata);
+                }
+
+                eventData.AddPayloadProperty("Exception", logEvent.Exception, _healthReporter, nameof(NLogInput));
+            }
+
+            // Inability to render the message, or any other LogEvent property, should not stop us from sending the event down the pipeline
+            try
+            {
+                eventData.AddPayloadProperty("Message", RenderLogEvent(Layout, logEvent), _healthReporter, nameof(NLogInput));
+            }
+            catch (Exception ex)
+            {
+                NLog.Common.InternalLogger.Warn(ex, NLogTargetTypeName + "(Name={0}): Failed to render event message", Name);
+                _healthReporter.ReportWarning($"{nameof(NLogInput)}: event message could not be rendered{Environment.NewLine}{ex.ToString()}");
+                eventData.AddPayloadProperty("Message", logEvent.FormattedMessage, _healthReporter, nameof(NLogInput));
+            }
+
+            // MessageTemplate is always present on Serilog events
+            eventData.AddPayloadProperty("MessageTemplate", logEvent.Message, _healthReporter, nameof(NLogInput));
+
+            if (IncludeMdc || IncludeMdlc)
+            {
+                CaptureLogEventContextProperties(logEvent, eventData);
+            }
+
+            if (IncludeEventProperties && logEvent.HasProperties)
+            {
+                CaptureLogEventPropertiesEventData(logEvent, eventData);
+            }
+
+            if (ContextProperties.Count > 0)
+            {
+                CaptureTargetContextEventData(logEvent, eventData);
+            }
+
+            return eventData;
+        }
+
+        private void CaptureLogEventContextProperties(LogEventInfo logEvent, EventData eventData)
+        {
+            var properties = GetContextProperties(logEvent);
+            if (properties != null)
+            {
+                foreach (var property in properties)
+                {
+                    try
+                    {
+                        eventData.AddPayloadProperty(property.Key, property.Value, _healthReporter, nameof(NLogInput));
+                    }
+                    catch (Exception ex)
+                    {
+                        NLog.Common.InternalLogger.Warn(ex, NLogTargetTypeName + "(Name={0}): Failed to render event property: {1}", Name, property.Key);
```

Comment:
This code that emits warnings about property rendering is very similar to code on lines 99-100, 161-162, 179-180 and 200-201. I suggest to create a helper method for this.